feat: Add autodoc for event annotations

bmtcril opened the pull request:
**Description:** Adds automatic documentation of annotated events for discoverability and ease of configuration. This is in draft until dependencies merge and release, so I can update requirements to the official version, I'll be adding event_key_fields for the rest of the events while that is in review. Generated docs look like this:

<img width="1374" alt="Screenshot 2024-02-20 at 1 11 21 PM" src="https://github.com/openedx/openedx-events/assets/112640379/c1a16ddc-cbfd-4c75-b2e7-7b6c3e57901e">


**Dependencies:** https://github.com/openedx/code-annotations/pull/98

**Installation instructions:** List any non-trivial installation
instructions.

**Testing instructions:**

1. Requirements are updated to use the branch of code-annotations necessary to run this so pulling this branch and running `make requirements` should suffice
2. Run `make docs`
3. Sphinx should run and pop open the generated docs in your browser. Navigate to reference -> events to see the content.

**Reviewers:**
- [ ] tag reviewer
- [ ] tag reviewer

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are squashed

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPI after tag-triggered build is
      finished.
- [ ] Delete working branch (if not needed anymore)

bmtcril commented:
Coverage issue is just because there is more code (comments), there is nothing to test in any of this.

mariajgrimaldi reviewed (commented):
This has my soft-approval. Thanks for doing this! We'll merge this once https://github.com/openedx/code-annotations/pull/98 is merged so we can the code-annotations release instead. Is that okay?

timmc-edx commented on docs/reference/events.rst:
```diff
@@ -0,0 +1,9 @@
+This is the list of Open edX events found in this repository.
+
+.. note::
+    Events can be create in other projects and plugins as well, but these default events are guaranteed to exist.
```
[nit] typo
```suggestion
    Events can be created in other projects and plugins as well, but these default events are guaranteed to exist.
```

timmc-edx reviewed (commented):
Looks good pending the code-annotations release!

bmtcril commented:
@timmc-edx @mariajgrimaldi this is ready for re-review. Changes since the last push:

- Using the release of code-annotations
- Ran make upgrade because of that
- Removed pii checks from Makefile and tox since they were broken after the removal of `.pii_annotations.yml`

mariajgrimaldi reviewed (approved):
LGTM. Thank you!
